src/main/scala/scala/slick/jdbc/StaticQuery.scala

```diff
@@ -60,72 +62,173 @@ class StaticQueryInvoker[-P, +R](val getStatement: String, pconv: SetParameter[P
 @deprecated("Use the new Action-based Plain SQL API from driver.api instead", "3.0")
 class SQLInterpolation(val s: StringContext) extends AnyVal {
   /** Build a SQLInterpolationResult via string interpolation */
-  def sql[P](param: P)(implicit pconv: SetParameter[P]) =
-    new SQLInterpolationResult[P](s.parts, param, pconv)
+  def sql(param: Any*) = macro SQLInterpolation.sqlImpl
   /** Build a StaticQuery for an UPDATE statement via string interpolation */
-  def sqlu[P](param: P)(implicit pconv: SetParameter[P]) = sql(param).asUpdate
+  def sqlu(param: Any*) = macro SQLInterpolation.sqluImpl
+  /** Build an Invoker for a statement with computed types via string interpolation */
+  def tsql(param: Any*) = macro SQLInterpolation.tsqlImpl
```

Comment:
Needs to be moved to ActionBasedSQLInterpolation (along with the implementing method)